src/Field.php

```diff
@@ -210,7 +210,7 @@ public function __construct(array $defaults = [])
     {
         foreach ($defaults as $key => $val) {
             if (is_array($val)) {
-                $this->{$key} = array_merge_recursive(is_array($this->{$key} ?? null) ? $this->{$key} : [], $val);
+                $this->{$key} = array_replace_recursive(is_array($this->{$key} ?? null) ? $this->{$key} : [], $val);
```

Comment:
Usecase:
```
// custom form control class defines
public $ui = ['form' => [CustomField::class, 'size' => 'small', 'color' => 'red']];

// in model i want to use this control with default size (small), but in green color
$this->addField('foo', ['ui' => ['form' => ['color' => 'green']]]);
```

So as result I need to recursively replace ui property not completely overwrite it.
Required result `ui => ['form' => [CustomField::class, 'size' => 'small', 'color' => 'green']]`